src/ws_server.rs

```diff
@@ -30,8 +34,40 @@ impl WsServer {
 
 impl<T: Controller> Handler for WsHandler<T> {
 
-    fn on_open(&mut self, _: Handshake) -> Result<()> {
+    fn on_open(&mut self, handshake: Handshake) -> Result<()> {
         info!("Hello new ws connection");
+
+        let resource = &handshake.request.resource()[..];
+        let url = Url::parse(&format!("http://fox.box{}", resource)).unwrap();
+
+        match url.query_pairs() {
+            Some(query_pairs) => {
+                let auth = query_pairs.iter()
+                    .find(|ref set| set.0.to_lowercase() == "auth")
+                    .map(|ref set| set.1.clone());
+
+                match auth {
+                    Some(ref token) => {
+                        if let Err(_) = AuthMiddleware::verify(token) {
+                            self.out.close_with_reason(
+                                ws::CloseCode::Error, "Authorization failed");
+                            return Ok(());
+                        }
+                    },
+                    _ => {
+                        self.out.close_with_reason(
+                            ws::CloseCode::Error, "Missing authorization");
+                        return Ok(());
+                    }
+                }
+            },
+            _ => {
+                self.out.close_with_reason(
+                    ws::CloseCode::Error, "Missing authorization");
+                return Ok(());
```

Comment:
I see it repeated three times and if you address https://github.com/fxbox/foxbox/pull/167#discussion_r55713780 we will have four "close with error message + return". But not a big deal.